lib/smart_proxy_pulp_plugin/disk_usage.rb

```diff
@@ -0,0 +1,52 @@
+module PulpProxy
+  class DiskUsage
+    include ::Proxy::Util
+    SIZE = { :kilobyte => 1_024, :megabyte => 1_048_576, :gigabyte => 1_073_741_824, :terabyte => 1_099_511_627_776 }
+
+    attr_reader :path, :stat, :size
+
+    def initialize(opts ={})
+      raise(::Proxy::Error::ConfigurationError, 'Unable to continue - must provide a path.') if opts[:path].nil?
+      @pathes_hash = path_hash(opts[:path])
+      @path = @pathes_hash.values.join(' ')
+      @size = SIZE[opts[:size]] || SIZE[:gigabyte]
+      @stat = {}
+      find_df
+      get_stat
+    end
+
+    def to_json
+      stat.to_json
+    end
+
+    private
+
+    attr_reader :command_path
+
+    def find_df
+      @command_path = which('df') || raise(::Proxy::Error::ConfigurationError, 'df command was not found unable to retrieve usage information.')
+    end
+
+    def command
+      "LANG=C #{command_path} -B#{size} #{path}"
```

Comment:
Sure, sorry for the delay, it's as simple as:

```ruby
>> Open3.popen3({"LC_ALL" => "C"}, "/usr/bin/df", "-h") {|i,o,e,t|
?>   puts o.read
>> }
```

The whole executable with list of args can be one array as well:

```ruby
Open3.popen3({"LC_ALL" => "C"}, ["/usr/bin/df", "-h"]) {|i,o,e,t|
  puts o.read
}
```